From 680e0cc26ce2c154e0bcd820762e3706240b5f26 Mon Sep 17 00:00:00 2001
From: Camillo Bruni <cbruni@chromium.org>
Date: Wed, 1 Feb 2023 20:33:40 +0000
Subject: [PATCH] [blink] Only use NodeCount in debug builds

It previously had no effect in non-DCHECK builds. This CL removes the
call overhead in release builds.

Bug: 808503
Change-Id: I9a007042f960eab4189755f1dccafa02ecab4461
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4209074
Commit-Queue: Camillo Bruni <cbruni@chromium.org>
Reviewed-by: Xiaocheng Hu <xiaochengh@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1100046}
---
 third_party/blink/renderer/core/dom/document.h | 2 ++
 third_party/blink/renderer/core/dom/node.cc    | 4 ++++
 2 files changed, 6 insertions(+)

--- a/third_party/blink/renderer/core/dom/document.h
+++ b/third_party/blink/renderer/core/dom/document.h
@@ -1634,12 +1634,14 @@
   static void SetForceSynchronousParsingForTesting(bool);
   static bool ForceSynchronousParsingForTesting();
 
+#if DCHECK_IS_ON()
   void IncrementNodeCount() { node_count_++; }
   void DecrementNodeCount() {
     DCHECK_GT(node_count_, 0);
     node_count_--;
   }
   int NodeCount() const { return node_count_; }
+#endif  // DCHECK_IS_ON()
 
   SnapCoordinator& GetSnapCoordinator();
   void PerformScrollSnappingTasks();
--- a/third_party/blink/renderer/core/dom/node.cc
+++ b/third_party/blink/renderer/core/dom/node.cc
@@ -2229,7 +2229,9 @@
          IsContainerNode());
   if (insertion_point.isConnected()) {
     SetFlag(kIsConnectedFlag);
+#if DCHECK_IS_ON()
     insertion_point.GetDocument().IncrementNodeCount();
+#endif
   }
   if (ParentOrShadowHostNode()->IsInShadowTree())
     SetFlag(kIsInShadowTreeFlag);
@@ -2248,7 +2250,9 @@
     ClearNeedsStyleInvalidation();
     ClearChildNeedsStyleInvalidation();
     ClearFlag(kIsConnectedFlag);
+#if DCHECK_IS_ON()
     insertion_point.GetDocument().DecrementNodeCount();
+#endif
   }
   if (IsInShadowTree() && !ContainingTreeScope().RootNode().IsShadowRoot())
     ClearFlag(kIsInShadowTreeFlag);
